File: RecyclerViews/LotsofLists/app/src/main/java/edu/cs4730/lotsoflists/myAdapter.java
package edu.cs4730.lotsoflists;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.util.List;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import edu.cs4730.lotsoflists.databinding.MyRowBinding;

/**
 * this adapter is very similar to the adapters used for listview, except a ViewHolder is required
 * see http://developer.android.com/training/improving-layouts/smooth-scrolling.html
 * except instead having to implement a ViewHolder, it is implemented within
 * the adapter.
 */

public class myAdapter extends RecyclerView.Adapter<myAdapter.ViewHolder> {

    private List<String> myList;
    private Context mContext;

    public myAdapter(List<String> myList, Context context) {
        this.myList = myList;
        this.mContext = context;
    }

    //a simple method to make sure the recyclerview knows I have changed the data in the list.
    public void newData(List<String> myList) {
        this.myList = myList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        MyRowBinding v = MyRowBinding.inflate(LayoutInflater.from(mContext), viewGroup, false);
        return new ViewHolder(v);
    }

    @Override
    public void onBindViewHolder(ViewHolder viewHolder, int i) {
        String entry = myList.get(i);
        viewHolder.viewBinding.Name.setText(entry);
        viewHolder.viewBinding.Name.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                TextView tv = (TextView) v;
                Toast.makeText(mContext, tv.getText(), Toast.LENGTH_SHORT).show();
            }
        });
        viewHolder.viewBinding.picture.setImageResource(R.drawable.phone);
    }

    @Override
    public int getItemCount() {
        return myList == null ? 0 : myList.size();
    }

    //the viewbinding now provides the references.
    public static class ViewHolder extends RecyclerView.ViewHolder {
        MyRowBinding viewBinding;

        public ViewHolder(MyRowBinding viewBinding) {
            super(viewBinding.getRoot());
            this.viewBinding = viewBinding;
        }
    }
}

File: RecyclerViews/LotsofLists_kt/app/src/main/java/edu/cs4730/lotsoflists_kt/myAdapter.kt
package edu.cs4730.lotsoflists_kt

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.TextView
import android.widget.Toast
import androidx.recyclerview.widget.RecyclerView
import edu.cs4730.lotsoflists_kt.databinding.MyRowBinding

/**
 * this adapter is very similar to the adapters used for listview, except a ViewHolder is required
 * see http://developer.android.com/training/improving-layouts/smooth-scrolling.html
 * except instead having to implement a ViewHolder, it is implemented within
 * the adapter.
 */
class myAdapter(
    private var myList: List<String>?,
    private val mContext: Context
) : RecyclerView.Adapter<myAdapter.ViewHolder>() {
    //a simple method to make sure the recyclerview knows I have changed the data in the list.
    fun newData(myList: List<String>?) {
        this.myList = myList
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val v = MyRowBinding.inflate(LayoutInflater.from(mContext), viewGroup, false)
        return ViewHolder(v)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, i: Int) {
        val entry = myList!![i]
        viewHolder.viewBinding.Name.text = entry
        viewHolder.viewBinding.Name.setOnClickListener { v ->
            val tv = v as TextView
            Toast.makeText(mContext, tv.text, Toast.LENGTH_SHORT).show()
        }
        viewHolder.viewBinding.picture.setImageResource(R.drawable.phone)
    }

    override fun getItemCount(): Int {
        return if (myList == null) 0 else myList!!.size
    }

    //the viewbinding now provides the references.
    class ViewHolder(var viewBinding: MyRowBinding) : RecyclerView.ViewHolder(
        viewBinding.root
    )
}